### PR TITLE
Allow agents using WebRequestConcern to disable URL encoding

### LINKED ISSUE #854: Wrong url_encode
I have an RSS Feed i'd like to scrape
```
https://foo.bar/bla/blub.aspx?GetRss?CategoryNr=12345
```

unfortunately, huginn replaces this to
```
https://foo.bar/bla/blub.aspx?GetRss%3FCategoryNr=12345
```
and the server returns an error...

### PR DESCRIPTION
**Pull request by dsander**
This should fix #854

### CODE REVIEW

File: app/concerns/web_request_concern.rb [hunk 1]
```diff
@@ -2,6 +2,18 @@
 require 'faraday_middleware'
 
 module WebRequestConcern
+  module DoNotEncoder
+    def self.encode(params)
+      params.map do |key, value|
+        value.present? ? "#{key}=#{value}" : "#{key}"
```
Comment (cantino) [inline app/concerns/web_request_concern.rb]: So `?key&key&key3` without values is valid URL syntax?

**Comment by cantino** [comment]
LGTM if it fixes the issue.

File: app/concerns/web_request_concern.rb (hunk 1 above)
Comment (dsander) [inline app/concerns/web_request_concern.rb]: Not sure, I just made sure @K1773R's [example](https://github.com/cantino/huginn/issues/854#issuecomment-118796264) works. In that case the `encode` function is passed `{GetRss?CategoryNr=39207' => nil}`. Rack parses your URL as `{"key"=>nil, "key3"=>nil}`

Comment (cantino) [inline app/concerns/web_request_concern.rb]: Sounds good.

There are some pretty odd webservers out there ;)

Comment (K1773R) [inline app/concerns/web_request_concern.rb]: @cantino no that isnt valid, tough i dont think its an issue allowing it.

Comment (knu) [inline app/concerns/web_request_concern.rb]: Shouldn't it be `value.nil? ? "#{key}" ? "#{key}=#{value}"`, considering `"".present?` is false?

Comment (dsander) [inline app/concerns/web_request_concern.rb]: Good point, that makes the results more coherent.

**Pull request merged**